src/Microsoft.Health.Fhir.Liquid.Converter/Filters/SegmentFilters.cs

```diff
@@ -17,6 +19,11 @@ public partial class Filters
     {
         public static Dictionary<string, Hl7v2Segment> GetFirstSegments(Hl7v2Data hl7v2Data, string segmentIdContent)
         {
+            if (hl7v2Data == null)
+            {
+                throw new TemplateLoadException(FhirConverterErrorCode.TemplateDataMismatch, "Incorrect template was passed in for the input data format.");
```

Comment:
Hi @pallar-ms 

I thought also that this would be the case, but no. Looks like that case is being caught most likely earlier before it gets to SegmentFilters. This is what I get when trying to reproduce hl7 not being passed in.

```json
{
  "error": {
    "code": "InvalidRequestBody",
    "message": "An error was found while validating the request body.",
    "innerError": {
      "code": "InvalidInputDataRequestValue",
      "message": "The InputDataString field value is invalid: The InputDataString field is required."
    }
  }
}
```

Reply:
Sure, but that's being handled in the api layer we have in convert PaaS. This is OSS project that has other consumers. In any case, it is just a suggestion based on the check that is quite generic.